pkg/controllers/nodeclass/status/controller.go

```diff
@@ -32,24 +32,29 @@ import (
 	"sigs.k8s.io/karpenter/pkg/utils/result"
 
 	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily"
 	"github.com/awslabs/operatorpkg/reasonable"
 )
 
-type nodeClassStatusReconciler interface {
+type Reconciler interface {
 	Reconcile(context.Context, *v1alpha2.AKSNodeClass) (reconcile.Result, error)
 }
 
 type Controller struct {
 	kubeClient client.Client
 
-	readiness *Readiness //TODO : Remove this when we have sub status conditions
+	k8sVersion *K8sVersionReconciler
```

Comment:
minor, style, feel free to ignore:

Even here I'd drop the `k8s` in favor of `Kubernetes`

Reply:
Agreed, since I updated the Reconciler name.